shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.mm

```diff
@@ -0,0 +1,132 @@
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.h"
+
+#import <QuartzCore/QuartzCore.h>
+#include <mutex>
+
+@interface FlutterThreadSynchronizer () {
+  std::mutex _mutex;
+  BOOL _shuttingDown;
+  CGSize _contentSize;
+  std::vector<dispatch_block_t> _scheduledBlocks;
+
+  BOOL _beginResizeWaiting;
+
+  // Used to block [beginResize:].
+  std::condition_variable _condBlockBeginResize;
+}
+
+@end
+
+namespace {
+/// Single use event (can only be signalled once)
+class Event {
```

Comment:
The event is only ever signalled once. It doesn't matter if it resets. I think it can be safely replaced with `AutoResetWaitableEvent`. I didn't do it originally because I didn't want to add another depenency on FML, but that's not a problem anymore so I'll go ahead and replace it.